src/plan/global.rs

```diff
@@ -360,6 +366,10 @@ pub trait Plan: 'static + Sync + Downcast {
             object
         );
     }
+
+    fn get_extra_header_bytes(&self) -> usize {
```

Comment:
To go for the first one, how can the binding know how many extra bytes it needs? What I can think of is to store ```header_reserved_in_bytes``` in ```markcompact_allocator``` as well. Then both ```markcompactspace``` and ```markcompact_allocator``` will have the same field.

Reply:
Can you store `header_reserved_in_bytes` as a constant in `MarkCompactSpace`? I feel that is possible, as the header size is a constant (1 word) and the alignment is a constant in `VMBinding`. The binding can just access this constant.

Reply:
It is a runtime constant rather than a compile-time constant. What I have got so far is something like 
```
struct MarkCompactAllocator {
  struct BumpAllocator bump_allocator;
  size_t header_reserved_in_bytes;
};
```
I need to figure out how to get that value

Reply:
It is a compile-time constant. Just you need a `VMBinding` type. You can do this (please check if this is correct):
```rust
trait VMBinding {
    const MAX_ALIGN: usize = 8;
}

const GC_HEADER_BYTES: usize = 8;

struct MarkCompactSpace<VM: VMBinding>(std::marker::PhantomData<VM>);
impl<VM: VMBinding> MarkCompactSpace<VM> {
    const HEADER_RESERVE: usize = if VM::MAX_ALIGN > GC_HEADER_BYTES {
        VM::MAX_ALIGN
    } else {
        GC_HEADER_BYTES
    }.next_power_of_two();
}

fn use_header_reserve<VM: VMBinding>() {
    println!("{}", MarkCompactSpace::<VM>::HEADER_RESERVE);
}
```